stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt

```diff
@@ -97,25 +97,49 @@ public class ChannelListViewModel(
         viewModelScope.launch {
             searchQuery.combine(queryConfig) { query, config -> query to config }
                 .collectLatest { (query, config) ->
-                    val filter = if (query.isNotEmpty()) {
-                        Filters.and(config.filters, Filters.autocomplete("name", query))
-                    } else {
-                        config.filters
-                    }
-
-                    val result = chatDomain.queryChannels(filter, config.querySort).await()
+                    val result = chatDomain.queryChannels(
+                        filter = createQueryChannelsFilter(config.filters, query),
+                        sort = config.querySort
+                    ).await()
 
                     if (result.isSuccess) {
                         observeChannels(controller = result.data(), searchQuery = query)
                     } else {
                         result.error().cause?.printStackTrace()
-                        channelsState =
-                            channelsState.copy(isLoading = false, channels = emptyList())
+                        channelsState = channelsState.copy(isLoading = false, channels = emptyList())
                     }
                 }
         }
     }
 
+    /**
+     * Creates a filter that is used to query channels.
+     *
+     * If the [searchQuery] is empty, then returns the original [filter] provided by the user.
+     * Otherwise, returns a wrapped [filter] that also checks that the channel name match the
+     * [searchQuery].
+     *
+     * @param filter The filter that was passed by the user.
+     * @param searchQuery The search query used to filter the channels.
+     * @return The filter that will be used to query channels.
+     */
+    private fun createQueryChannelsFilter(filter: FilterObject, searchQuery: String): FilterObject {
```

Comment:
This is really nice! Well done, I'm glad we can support this and optimize it to some extent (with the notExists check).

I've tested it and it works correctly, so all seems to be there! :]